fitbenchmarking/controllers/base_matlab_controller.py

```diff
@@ -0,0 +1,78 @@
+"""
+Implements base class for the matlab fitting software controllers.
+"""
+
+try:
+    from tempfile import TemporaryDirectory
+except ImportError:
+    from backports.tempfile import TemporaryDirectory
+import os
+try:
+    import dill
+    import_success = True
+except ImportError:
+    import_success = False
+
+from fitbenchmarking.controllers.base_controller import Controller
+from fitbenchmarking.utils.exceptions import MissingSoftwareError
+
+
+class BaseMatlabController(Controller):
+    """
+    Base class for matlab fitting software controllers
+    """
+
+    def __init__(self, cost_func):
+        """
+        Initialise anything that is needed specifically for matlab
+        fitting software
+        """
+        super().__init__(cost_func)
+
+        if not import_success:
+            raise MissingSoftwareError('Requirements are missing for Matlab '
+                                       'fitting, module "dill" is reuqired.')
+
+        self.initial_params_mat = None
+
+    def setup(self):
+        raise NotImplementedError
+
+    def jacobian_information(self):
+        raise NotImplementedError
+
+    def fit(self):
+        raise NotImplementedError
+
+    def cleanup(self):
+        raise NotImplementedError
+
+    def py_to_mat(self, func, eng):
```

Comment:
Another minor comment is that this will raise the pylint refactor warning about no self use.

e can overcome that by making it a static_method (https://docs.python.org/3/library/functions.html#staticmethod)